Android/app/src/main/java/io/github/project_travel_mate/destinations/description/WeatherActivity.java

```diff
@@ -0,0 +1,334 @@
+                                            LinearLayoutManager.HORIZONTAL, false));
+
+                                    WeatherAdapter adapter = new WeatherAdapter(WeatherActivity.this, mWeatherList);
+                                    forecast_list.setAdapter(adapter);
+                                }
+                            }
+                            //cancel the loading animation once the data is fetched
+                            cancelAnimation();
+
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                            //display no internet connection error
+                            networkError();
+                        }
+                    } else {
+                        //display animation for no data returned from the API call
+                        emptyListAnimation();
+                    }
+                });
+            }
+
+        });
+    }
+
+    private int fetchDrawableFileResource(String iconUrl, int code) {
+        //formulate the vector drawable file name from the local json file using the icon url & code from the API call
+        String imageDrawable = "wi_";
+        String time = iconUrl.substring(iconUrl.lastIndexOf("/") + 1);
+        imageDrawable += time.contains("d") ? "day" : "night";
+        String suffix = getSuffix(code);
+        imageDrawable += "_" + suffix;
+
+        //return the vector drawable resource id
+        return getResources().getIdentifier(imageDrawable, "drawable", "io.github.project_travel_mate");
+    }
+
+    /**
+     * parses the icons.json file which contains the weather condition codes and descriptions
+     * required to fetch the right weather icon to display
+     *
+     * @param code weather condition code
+     * @return weather condition description
+     */
+    private String getSuffix(int code) {
+        String json;
+        String cond = "";
+        try {
+            InputStream is = getAssets().open("icons.json");
+            int size = is.available();
+            byte[] buffer = new byte[size];
+            is.read(buffer);
+            is.close();
+            json = new String(buffer, "UTF-8");
+
+            JSONObject jsonObject = new JSONObject(json);
+            if (jsonObject.has(String.valueOf(code))) {
+                JSONObject object = jsonObject.getJSONObject(String.valueOf(code));
+                cond = object.getString("icon");
+            }
+
+        } catch (JSONException ex) {
+            emptyListAnimation();
+        } catch (IOException e) {
+            networkError();
+        }
+        return cond;
+    }
+
+    /**
+     * called to get the days of the week needed to display the forecast
+     *
+     * @param index day's index
+     * @return current day of the week as a String
+     */
+    private String getDayOfWeek(int index) {
```

Comment:
add a new file in `utils` named `WeatherUtils`, and move 
`fetchDrawableFileResource` 
`getSuffix`
`getDayOfWeek`
in that file